### PR TITLE
Add generateMapToFullTableNames in TableMappingService

### PR DESCRIPTION
**Pull request by hsaraogi**
Duplicate of #1486 based on 0.25.x branch

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/1568)

### CODE REVIEW

File: atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/impl/AbstractTableMappingService.java [hunk 1]
```diff
@@ -94,31 +94,32 @@ private TableReference getFullTableName(TableReference shortTableName) {
     private final ConcurrentHashMap<TableReference, Boolean> unmappedTables = new ConcurrentHashMap<>();
 
     @Override
-    public Set<TableReference> mapToFullTableNames(Set<TableReference> tableRefs) {
-        Set<TableReference> newSet = Sets.newHashSet();
+    public Map<TableReference, TableReference> generateMapToFullTableNames(Set<TableReference> tableRefs) {
+        Map<TableReference, TableReference> shortNameToFullTableName = Maps.newHashMapWithExpectedSize(
```
Comment (schlosna) [inline atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/impl/AbstractTableMappingService.java]: nit: might as well use an `ImmutableMap.builder()`

**Comment by rhero** [comment]
@cholve @tkleingarn - Did you guys want this fix backported to 0.25.x still? I'm not sure if the recent perf regressions change the target version for the internal LTS release.

**Comment by cholve** [comment]
@rhero yes, I'm pretty sure we still want this backported to 0.25 because we'd like to get this into the previous internal quarterly release.  As I understand it the recent perf regression only affects brand new installs of the internal product on the internal quarterly release, which means we can't bump the Atlas version on those installers anyway and we're planning on tackling the fix for the perf regression with an internal upgrade task outside of AtlasDB.  @tkleingarn, does that sound right to you?

**Comment by cholve** [comment]
@hsaraogi it looks like the circle CI build is failing to download something related to setting up the build environment.  It doesn't look like I have any permissions to retrigger.  Do you know what might be happening?

Error from circle build:
```+ curl -O http://ftp.de.debian.org/debian/pool/main/m/moreutils/moreutils_0.59-1_amd64.deb
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

100   336  100   336    0     0   1256      0 --:--:-- --:--:-- --:--:--  3326
+ ar p moreutils_0.59-1_amd64.deb data.tar.xz
+ sudo tar xJ --strip-components=3 -C /usr/bin/ ./usr/bin/ts
ar: moreutils_0.59-1_amd64.deb: File format not recognized
xz: (stdin): File format not recognized
tar: Child returned status 1
tar: Error is not recoverable: exiting now

./scripts/circle-ci/download-moreutils.sh returned exit code 2```

**Comment by cholve** [comment]
@hsaraogi any chance you have any insight into the circle error?

**Comment by rhero** [comment]
Looks like something funky with circle. These errors showed up on develop about 25 days ago. For reference, the internal circle builds see this:

```
sudo service clamav-freshclam stop
clamav-freshclam: unrecognized service

sudo service clamav-freshclam stop returned exit code 1

Action failed: sudo service clamav-freshclam stop
```

The current circle.yml on develop looks quite a bit different from the one on 0.25 after Ben did a bunch of improvements to builds.

@SerialVelocity - do the above build failures ring a bell? Do you know what changes we would need to make to the 0.25.x branch in order to get builds passing?

**Comment by SerialVelocity** [comment]
Anything that far back should rely on the external CircleCI status since it hasn't got the internal CircleCI fixes that are needed. Some of the fixes needed for internal builds to pass are in #1319 but it probably isn't all of them.

**Comment by rhero** [comment]
@hsaraogi did you want a review of any sort on this? If it's just a straight backport, then we can probably merge this now that builds are passing.

**Comment by cholve** [comment]
It looked like DavidS did a quick review earlier on, and @hsaraogi did a review of the original change a few weeks back.  I think we should be good to merge, right?

**Comment by rhero** [comment]
Cool, that reasoning works for me. I'm going to merge this and tag it for you.

**Pull request merged**